src/sentry/snuba/discover.py

```diff
@@ -427,10 +442,18 @@ def timeseries_query(selected_columns, query, params, rollup, referrer=None):
         op="discover.discover", description="timeseries.filter_transform"
     ) as span:
         span.set_data("query", query)
-        snuba_filter, _ = get_timeseries_snuba_filter(selected_columns, query, params, rollup)
+        snuba_filter, _ = get_timeseries_snuba_filter(selected_columns, query, params)
 
     with sentry_sdk.start_span(op="discover.discover", description="timeseries.snuba_query"):
         result = raw_query(
+            # Hack cause equations on aggregates have to go in selected columns instead of aggregations
+            selected_columns=[
+                column
+                for column in snuba_filter.selected_columns
+                if isinstance(column, list)
+                and len(column) == 3
+                and column[-1].startswith("equation")
```

Comment:
I understand the intent of this right now but I feel this is going to be confusing down the road. A helper and/or some comments explaining what the JSON for an equation looks like will help here.

Reply:
Gonna go with a comment for now, cause SnQL will let us directly check the alias instead of these 3 checks

Reply:
Should this check be a regex match for something like `equation\[\d+\]` since it's possible a tag is prefixed with `equation`?

Reply:
Good point, while its possible a tag is prefixed with equation, its not possible for a tag to contain `[]`,  so I'll do a `startswith('equation[')` instead